### PR TITLE
Fix parsing anonymous interface and structs

### LINKED ISSUE #12: Error parsing fields with anonymous interface and structs
**Describe the bug**

While parsing anonymous types, being `interface{}` or `struct{}`, the lib does not handle it properly.

For interfaces, a "const" `InterfaceRefType` is returned.

For structs, an "unexpected type error" is returned.

```go
type User struct {
	ID int64 `json:"id"`
	Name  string `json:"name"`
	Email string `json:"email"`
	Data  interface{} // Problem here
	Data2 struct { // Here too
		Name string
	}
}
```

### PR DESCRIPTION
**Pull request by jamillosantos**
### :bug: Bugs

* Fix parsing fields with anonymous interfaces declarations;
* Fix parsing fields with anonymous structs declarations;

---

Fix #12

### CODE REVIEW

File: data/models10.sample.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Package models is a test
+package models
+
```
Comment (Jeconias) [inline data/models10.sample.go]: Add import fmt

File: data/models6.sample.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package models
+
```
Comment (Jeconias) [inline data/models6.sample.go]: Add import fmt.

File: data/models10.sample.go (hunk 1 above)
Comment (jamillosantos) [reply data/models10.sample.go]: That is an old commit and I have not changed it.

File: data/models6.sample.go (hunk 2 above)
Comment (jamillosantos) [reply data/models6.sample.go]: That is an old commit and I have not changed it.

**Review by Jeconias — APPROVED** [review]

**Pull request merged**